src/tokenizer.rs

```diff
@@ -1229,15 +1229,31 @@ impl<'a> Tokenizer<'a> {
                 // operators
                 '-' => {
                     chars.next(); // consume the '-'
-                    match chars.peek() {
-                        Some('-') => {
-                            chars.next(); // consume the second '-', starting a single-line comment
+
+                    if let Some('-') = chars.peek() {
+                        let mut is_comment = true;
+                        if self.dialect.requires_whitespace_to_start_comment() {
+                            // MySQL requires a space after the -- for a single-line comment
+                            // Otherwise it's interpreted as two minus signs
+                            // e.g. UPDATE account SET balance=balance--1
+                            //      WHERE account_id=5752;
```

Comment:
Can we move the example to the doc string in the dialect method? That would help clarify the behavior to folks without having to dig deeper in the docs or the parser code.
Then for the rest of the comment here we can probably skip since we've already documented as much in the dialect method